packages/database/src/models/User.ts

```diff
@@ -265,52 +266,56 @@ export class User extends Model {
     return false;
   }
 
-  async checkCanAccessAllFacilities() {
-    const restrictUsersToFacilities = await this.sequelize.models.Setting.get(
-      'auth.restrictUsersToFacilities',
-    );
-    if (!restrictUsersToFacilities) return true;
-    if (this.isSuperUser()) return true;
-    // Allow for roles that have access to all facilities configured via permissions
-    // (e.g. a custom "AdminICT" role)
-    if (await this.hasPermission('login', 'Facility')) return true;
-    return false;
-  }
-
   async allowedFacilities() {
-    const canAccessAllFacilities = await this.checkCanAccessAllFacilities();
-    if (canAccessAllFacilities) {
-      return CAN_ACCESS_ALL_FACILITIES;
-    }
+    const { Facility, Setting } = this.sequelize.models;
+
+    if (this.isSuperUser()) return CAN_ACCESS_ALL_FACILITIES;
+
+    const restrictUsersToFacilities = await Setting.get('auth.restrictUsersToFacilities');
+    const hasLoginPermission = await this.hasPermission('login', 'Facility');
 
+    // Get user's linked facilities
```

Comment:
Feels like there's a performance optimisation in here where you check if there are any sensitive facilities at all, and return `CAN_ACCESS_ALL_FACILITIES` if not, before reloading their facilities. That would improve performance here, but also any time we check their list of facilities (because it can just match that const rather than looking through a potentially long list)